driver/src/main/scala/core/commands/commands.scala

```diff
@@ -592,7 +592,7 @@ case class DeleteIndex(
 
   object ResultMaker extends BSONCommandResultMaker[Int] {
     def apply(document: BSONDocument) =
-      CommandError.checkOk(document, Some("deleteIndexes")).toLeft(document.getAs[BSONDouble]("nIndexesWas").map(_.value.toInt).get) }
+      CommandError.checkOk(document, Some("deleteIndexes")).toLeft(document.getAs[BSONInteger]("nIndexesWas").map(_.value.toInt).get) }
```

Comment:
It may be preferable to use `BSONNumberLike` instead, then converting it to `Int`.  Then it will work in any case.